reference/src/glossary.md

```diff
@@ -190,8 +190,9 @@ guarantee that `Option<&mut T>` has the same size as `&mut T`.
 
 While all niches are invalid bit-patterns, not all invalid bit-patterns are
 niches. For example, the "all bits uninitialized" is an invalid bit-pattern for
-`&mut T`, but this bit-pattern cannot be used by layout optimizations, and is not a
-niche.
+`&mut T`, but this bit-pattern cannot be used by layout optimizations, and is not a niche.
+
+It is a surprisingly common misconception that niches can occur in [padding] bytes. They cannot: A niche representation must be invalid for `T`. But a padding byte must be irrelevant to the value of `T`. It follows that if you take a niche representation of `T`, and change any of the padding bytes to any other values, then the result must still be a niche representation of `T`. If a niche were contained entirely in padding, that would mean that `T` was entirely niches and, consequently, uninhabited.
```

Comment:
Please line-break this like the rest of the file.

Also I have to say I find the 2nd part (starting with "it follows") of this quite hard to follow, even though I understand that padding cannot be niches. ;)  The first part already basically says that padding is the opposite of a niche: padding is *irrelevant* for the represented value, whereas niches must definitely *not* represent a value. Padding is allowed to change arbitrarily any time, so obviously we cannot store any relevant information there.

Might be worth mentioning that we *could* define a representation where the gaps between fields are guaranteed 0. *That* kind of padding could be used as a niche. (I thought I had created an issue proposing this somewhere but I cannot find it right now...) The contrast between that kind of guaranteed-0-padding that can have a niche, and the normal ignored-padding that cannot, should also help drive home the point.

Reply:
Guaranteed-0 padding does not meet the current definition of padding, though, so that's possibly a bit academic. I will clean this up per comment, but I'm not sure about introducing the concept of guaranteed-0 padding given that we don't already have it.